src/components/interfaces.ts

```diff
@@ -9,3 +9,10 @@ export type Scale = "s" | "m" | "l";
 export type Status = "invalid" | "valid" | "idle";
 export type ThemeClass = "calcite-theme-light" | "calcite-theme-dark" | "calcite-theme-auto";
 export type Width = "auto" | "half" | "full";
+
+export interface CalciteFormComponent {
+  effectiveLabel: HTMLCalciteLabelElement;
+  effectiveLabelClickHandler: () => void;
+  connectEffectiveLabel: () => void;
+  disconnectEffectiveLabel: () => void;
```

Comment:
Some other properties we could add here:

- disabled
- label
- loading 
- name
- readOnly 
- required 
- value

Reply:
I think we can exclude name, label, loading, readOnly, required. Not all form components have one of those.

Reply:
There's no 'loading', 'readOny', 'required' for something like checkbox.